entity/templates/src/main/java/package/web/rest/_EntityResource.java

```diff
@@ -40,23 +59,32 @@
     private <%= entityClass %>Repository <%= entityInstance %>Repository;<% if (searchEngine == 'elasticsearch') { %>
 
     @Inject
-    private <%= entityClass %>SearchRepository <%= entityInstance %>SearchRepository;<% } %>
+    private <%= entityClass %>SearchRepository <%= entityInstance %>SearchRepository;<% } %><% if (generateDto == 'yes') { %>
+
+    @Inject
+    private <%= entityClass %>Mapper <%= entityInstance %>Mapper;
 
+    @Inject
+    private <%= entityClass %>DetailMapper <%= entityInstance %>DetailMapper;<% } %>
+    
     /**
      * POST  /<%= entityInstance %>s -> Create a new <%= entityInstance %>.
      */
     @RequestMapping(value = "/<%= entityInstance %>s",
             method = RequestMethod.POST,
             produces = MediaType.APPLICATION_JSON_VALUE)
-    @Timed
-    public ResponseEntity<Void> create(<% if (validation) { %>@Valid <% } %>@RequestBody <%= entityClass %> <%= entityInstance %>) throws URISyntaxException {
-        log.debug("REST request to save <%= entityClass %> : {}", <%= entityInstance %>);
-        if (<%= entityInstance %>.getId() != null) {
+    @Timed<% if (databaseType == 'sql') { %>
+    @Transactional(readOnly = false)<% } %>
```

Comment:
I think the same ! I have a follow up change, where I've moved all the transactional stuff to a service. I didn't add to this commit because I thought, the commit is already too big and complex, and you've stated, that you don't prefer to generate a separate service per entity, so it would be even harder time to sell to merge it :)